src/classes/ParallelProcessing.py

```diff
@@ -48,28 +55,42 @@ def run(self):
             answer = self.screenStocks(*(next_task))
             self.task_queue.task_done()
             self.result_queue.put(answer)
-            
 
     def screenStocks(self, executeOption, reversalOption, daysForLowestVolume, minRSI, maxRSI, respBullBear, insideBarToLookback, totalSymbols,
-                    configManager, fetcher, screener, candlePatterns, stock):
-        global screenCounter, screenResultsCounter
+                     configManager, fetcher, screener, candlePatterns, stock):
         screenResults = pd.DataFrame(columns=[
             'Stock', 'Consolidating', 'Breaking-Out', 'MA-Signal', 'Volume', 'LTP', 'RSI', 'Trend', 'Pattern'])
         screeningDictionary = {'Stock': "", 'Consolidating': "",  'Breaking-Out': "",
                                'MA-Signal': "", 'Volume': "", 'LTP': 0, 'RSI': 0, 'Trend': "", 'Pattern': ""}
-        saveDictionary = {'Stock': "", 'Pattern': "", 'Consolidating': "", 'Breaking-Out': "",
+        saveDictionary = {'Stock': "", 'Consolidating': "", 'Breaking-Out': "",
                           'MA-Signal': "", 'Volume': "", 'LTP': 0, 'RSI': 0, 'Trend': "", 'Pattern': ""}
 
         try:
-            data = fetcher.fetchStockData(stock,
-                                                configManager.period,
-                                                configManager.duration,
-                                                self.proxyServer,
-                                                self.screenResultsCounter,
-                                                self.screenCounter,
-                                                totalSymbols)
-                                                
-            fullData, processedData = screener.preprocessData(data, daysToLookback=configManager.daysToLookback)
+            if (self.stockDict.get(stock) is None) or (configManager.cacheEnabled == False) or self.isTradingTime:
+                data = fetcher.fetchStockData(stock,
+                                              configManager.period,
+                                              configManager.duration,
+                                              self.proxyServer,
+                                              self.screenResultsCounter,
+                                              self.screenCounter,
+                                              totalSymbols)
+                if configManager.cacheEnabled == True and not self.isTradingTime:
+                    self.stockDict[stock] = data.to_dict('split')
```

Comment:
Is it possible to use `pickle` to save this data on disk so it will persist in aftermarket hours even after the program restarts?
Possible to `pickle` entire dictionary in `screenipy.py` after displaying result table.

Reply:
Yes, saving data on the disk will work. 👍